wordpress-mobile/WordPress-iOS
# Site Settings in SwiftUI: Related Posts

HACK week: update settings screen to use SwiftUI

- Replace `RelatedPostsSettingsViewController` (Objective-C) and add `SiteSettingsRelatedPostsView` (SwiftUI)
- Contains the save fix ([#20894](https://github.com/wordpress-mobile/WordPress-iOS/pull/20894))
- Fixes the layout issues (clipped views)

Before → After

<img width="302" alt="settings-issues" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/85e3bbce-36fa-4cd4-bec1-79c23d7c9355"> <img width="302" alt="settings-issues-fixed" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/ecc35bca-8374-4763-a48c-fe1e32483b3f">


## To test:

- Open Site Settings / Related Posts
- Update the settings
- Verify that it matches the previous behavior (and addresses layout issues)
- Verify that after changing settings, it's immediately updated on the site

## Regression Notes
1. Potential unintended areas of impact: Site Settings / Related Posts
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual testing
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:


UI Changes testing checklist:
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout) - [Evidence](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/b1e31c6e-8f56-44dd-822d-1978fb9bfb79)
